pkg/k8shandler/deployment.go

```diff
@@ -554,48 +563,63 @@ func (node *deploymentNode) isChanged() bool {
 		nodeContainer := node.self.Spec.Template.Spec.Containers[index]
 		desiredContainer := desired.Spec.Template.Spec.Containers[index]
 
-		if nodeContainer.Resources.Requests == nil {
-			nodeContainer.Resources.Requests = v1.ResourceList{}
-		}
+		var updatedContainer v1.Container
+		updatedContainer, changed = updateResources(node, nodeContainer, desiredContainer)
 
-		if nodeContainer.Resources.Limits == nil {
-			nodeContainer.Resources.Limits = v1.ResourceList{}
-		}
+		node.self.Spec.Template.Spec.Containers[index] = updatedContainer
+	}
 
-		// check that both exist
+	return changed
+}
 
-		if nodeContainer.Image != desiredContainer.Image {
-			logrus.Debugf("Resource '%s' has different container image than desired", node.self.Name)
-			nodeContainer.Image = desiredContainer.Image
-			changed = true
-		}
+//updateResources for the node; return true if an actual change is made
+func updateResources(node *deploymentNode, nodeContainer, desiredContainer v1.Container) (v1.Container, bool) {
+	changed := false
+	if nodeContainer.Resources.Requests == nil {
+		nodeContainer.Resources.Requests = v1.ResourceList{}
+	}
 
-		if desiredContainer.Resources.Limits.Cpu().Cmp(*nodeContainer.Resources.Limits.Cpu()) != 0 {
-			logrus.Debugf("Resource '%s' has different CPU limit than desired", node.self.Name)
-			nodeContainer.Resources.Limits[v1.ResourceCPU] = *desiredContainer.Resources.Limits.Cpu()
-			changed = true
-		}
-		// Check memory limits
-		if desiredContainer.Resources.Limits.Memory().Cmp(*nodeContainer.Resources.Limits.Memory()) != 0 {
-			logrus.Debugf("Resource '%s' has different Memory limit than desired", node.self.Name)
-			nodeContainer.Resources.Limits[v1.ResourceMemory] = *desiredContainer.Resources.Limits.Memory()
-			changed = true
-		}
-		// Check CPU requests
-		if desiredContainer.Resources.Requests.Cpu().Cmp(*nodeContainer.Resources.Requests.Cpu()) != 0 {
-			logrus.Debugf("Resource '%s' has different CPU Request than desired", node.self.Name)
-			nodeContainer.Resources.Requests[v1.ResourceCPU] = *desiredContainer.Resources.Requests.Cpu()
-			changed = true
+	if nodeContainer.Resources.Limits == nil {
+		nodeContainer.Resources.Limits = v1.ResourceList{}
+	}
+
+	// check that both exist
+
+	if nodeContainer.Image != desiredContainer.Image {
```

Comment:
why are we checking `Image` in `updateResources`?

Reply:
This was grabbed in the refactor.  I'll just move it out.